### PR TITLE
doc/cephadm: update compatability and stability page

### PR DESCRIPTION
**Pull request by adk3798**
Removing the podman compatability chart as there have been not been any cephadm <-> podman compatability issues since podman 3.0 and the initial quincy release so the table isn't needed any more on main branch.

Additionally updates the list of things that are considered "unstable" or under development

Fixes: https://tracker.ceph.com/issues/68113

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: doc/cephadm/compatibility.rst [hunk 1]
```diff
@@ -55,11 +14,13 @@ open a tracker issue under the Orchestrator component (https://tracker.ceph.com/
 
 Cephadm support remains under development for the following features:
 
-- ceph-exporter deployment
 - stretch mode integration
-- monitoring stack (moving towards prometheus service discover and providing TLS)
+- monitoring stack (moving towards prometheus service discovery and providing TLS)
```
Comment (anthonyeleven) [inline doc/cephadm/compatibility.rst:18]: Prometheus capitalized please.

**Review by anthonyeleven — COMMENTED** [review]
I hesitate to remove the table.  Not everyone has the ability to run the latest releases.

**Review by ktdreyer — APPROVED** [review]

**Comment by ktdreyer** [comment]
> I hesitate to remove the table. Not everyone has the ability to run the latest releases.
It's staying for older Ceph releases.

**Comment by adk3798** [comment]
> I hesitate to remove the table. Not everyone has the ability to run the latest releases.

The idea was that the table would still exist in the octopus, pacific, quincy branches (and reef and squid I guess since I won't backport this) but there's no need for it in the T release that main will become.

File: doc/cephadm/compatibility.rst (hunk 1 above)
Comment (adk3798) [reply doc/cephadm/compatibility.rst]: done

**Pull request closed without merging**